helm-chart/renku-notebooks/templates/hpa.yaml

```diff
@@ -0,0 +1,14 @@
+{{- if .Values.autoscaling.enabled }}
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: {{ template "notebooks.fullname" . }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ template "notebooks.fullname" . }}
+  minReplicas: {{ .Values.autoscaling.minReplicas }}
```

Comment:
Do you know if the HPA guarantees that the replicas are spread out over at least two nodes?

Reply:
pretty sure it just hands the request to the scheduler - you'd need to have anti-affinity set separately. This isn't about HA I guess

Reply:
I think it does

Reply:
You're right @pameladelgado. The scheduler in its default settings will try to spread out pods, but it won't guarantee anything (it's just entering the big priority scoring function): 
```
SelectorSpreadPriority: Spreads Pods across hosts, considering Pods that belong to the same Service, StatefulSet or ReplicaSet.
```